hack/scripts/sync-prow.sh

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+check_deps() {
+  if ! which git-filter-repo > /dev/null 2>&1; then
+    echo "please install the git-filter-repo tool per https://github.com/newren/git-filter-repo"
+    exit 1
+  fi
+}
+
+check_deps
+
+old_prow="$PWD/test-infra"
+new_prow="$PWD/prow"
+
+# Clone the old prow repo.
+if [[ ! -d "$old_prow" ]]; then
+    git clone https://github.com/kubernetes/test-infra.git
```

Comment:
Another way to make this perhaps more obvious is to tell git to use a directory name directly, like this:

```
old_prow="$PWD/old_prow"
git clone https://github.com/kubernetes/test-infra.git $old_prow || true
```

That way it can just be a 1-liner, and `old_prow` will just be named `old_prow` to match the git remote name we give it on line 37.